nucliadb/nucliadb/export_import/utils.py

```diff
@@ -71,13 +72,39 @@ async def import_broker_message(
 ) -> None:
     bm.kbid = kbid
     partition = context.partitioning.generate_partition(kbid, bm.uuid)
-    for import_bm in [get_writer_bm(bm), get_processor_bm(bm)]:
+    for pb in [get_writer_bm(bm), get_processor_bm(bm)]:
+        await transaction_commit(context, pb, partition)
+
+
+async def transaction_commit(
+    context: ApplicationContext, bm: writer_pb2.BrokerMessage, partition: int
+) -> None:
+    """
+    Try to send the broker message over nats. If it's too big, upload
+    it to blob storage and over nats only send a reference to it.
+    """
+    try:
         await context.transaction.commit(
-            import_bm,
+            bm,
             partition,
             wait=False,
             target_subject=Streams.INGEST_PROCESSED.subject,
         )
+    except nats.errors.MaxPayloadError:
```

Comment:
i sthere a way for us to know that the payload is too big *before* we send it? so we would pick the right strategy without having to fail first

Reply:
however, one caveat of this is that we'll need to serialize the broker message every time to compute its lenght

Reply:
I think it's OK to leave it as it was... It also follows more the “Easier to Ask Forgiveness Than Permission” style